src/apps/activity_assignments/tests/test_unassignments.py

```diff
@@ -0,0 +1,765 @@
+import http
```

Comment:
you duplicate all tests for assignments, but you are not creating a new entitity, you are updating "is_deleted" field for already existing entities "assignments"

Reply:
The tests create the assignment first and then it checks if the unassignment was actually executed for example: 


        # Step 1: Create an assignment in applet_two
        assignments_create = ActivitiesAssignmentsDelete(
            assignments=[
                ActivityAssignmentCreate(
                    activity_id=applet_two.activities[0].id,
                    respondent_subject_id=tom_applet_one_subject.id,
                    target_subject_id=tom_applet_one_subject.id,
                )
            ]
        )
        await client.post(
            self.activities_assign_unassign_applet.format(applet_id=applet_two.id),
            data=assignments_create,
        )

        # Step 2: Attempt to unassign it using the wrong applet_id (applet_one)
        assignment_delete = ActivitiesAssignmentsDelete(
            assignments=[
                ActivityAssignmentCreate(
                    activity_id=applet_two.activities[0].id,
                    respondent_subject_id=tom_applet_one_subject.id,
                    target_subject_id=tom_applet_one_subject.id,
                )
            ]
        )
        
       How would be the correct way to check for unassignments. I made a comment somewhere asking If I should create the assignments as part of an @pytest.fixture but I ran into some async problems while doing so and I kept this format of creating and then deleting in the meantime

Reply:
Returning None and testing for 
        assert unassign_response.status_code == http.HTTPStatus.NO_CONTENT
Errors are handled using the _calidation function and the new error classes
Commited in last changes